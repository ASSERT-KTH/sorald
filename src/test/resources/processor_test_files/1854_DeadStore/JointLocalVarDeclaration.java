/*
Joint declarations are tricky in Spoon as they are modelede as separate elements with the same
source position. This test case tests that Sorald can find the correct local variable in a few
different configurations of joint declarations.
 */

public class JointLocalVarDeclaration {

    // the first variable in a joint declaration is a dead store
    public void firstIsDeadStore() {
        int x = 2, y = 3, z = 4; // Noncompliant
        System.out.println(y);
        System.out.println(z);
    }

    // the second variable in a joint declaration is a dead store
    public void secondIsDeadStore() {
        int x = 2, y = 3, z = 4; // Noncompliant
        System.out.println(x);
        System.out.println(z);
    }

    // the third variable in a joint declaration is a dead store
    public void thirdIsDeadStore() {
        int x = 2, y = 3, z = 4; // Noncompliant
        System.out.println(x);
        System.out.println(y);
    }
}
