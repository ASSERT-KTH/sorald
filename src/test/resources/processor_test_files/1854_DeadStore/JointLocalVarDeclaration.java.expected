/*
Joint declarations are tricky in Spoon as they are modeled as separate elements with the same
source position. This test case tests that Sorald can find the correct local variable in a few
different configurations of joint declarations.
 */

public class JointLocalVarDeclaration {

    // the first variable in a joint declaration is a dead store
    public void firstIsDeadStore() {
        int y = 3;
        int z = 4;
        System.out.println(y);
        System.out.println(z);
    }

    // the second variable in a joint declaration is a dead store
    public void secondIsDeadStore() {
        int x = 2;
        int z = 4;
        System.out.println(x);
        System.out.println(z);
    }

    // the third variable in a joint declaration is a dead store
    public void thirdIsDeadStore() {
        int x = 2;
        int y = 3;
        System.out.println(x);
        System.out.println(y);
    }

    // no whitespace to delimit the identifiers
    public void noWhitespaceBetweenIdentAndComma() {
        int x = 2;
        int z = 4;
        System.out.println(x);
        System.out.println(z);
    }
}
